### DISCUSSION TITLE
[Feature] Check media hash before uploading

### DISCUSSION THREAD

**Discussion by SerAlbi** [immich-app/immich#8840] [Feature Request]
### I have searched the existing feature requests to make sure this is not a duplicate request.

- [X] Yes

### The feature

My main photo backup folder on PC already contains many of the photos i have on my mobile phone. If i upload the PC backup folder on immich and then also enable photo backup on the mobile app, basicalli 90% of my library needs to be re-uploaded to immich for no reason because 90% of the photos will be discarded as duplicates.

Immich should generate the photo's hash BEFORE uploading, then it should check with the immich server for duplicates and then uploading if no match is found. 
This would save a lot of bandwidth and time for big libraries and i see no reason why the duplicate's search is only done after uploading..

### Platform

- [X] Server
- [X] Web
- [X] Mobile

**Comment by bo0tzz** [comment]
#1553
